js/wp-seo-wordpress-scraper.js

```diff
@@ -0,0 +1,394 @@
+    this.formattedData.queue = ["wordCount",
+        "keywordDensity",
+        "subHeadings",
+        "stopwords",
+        "fleschReading",
+        "linkCount",
+        "imageCount",
+        "urlKeyword",
+        "urlLength",
+        "metaDescription",
+        "pageTitleKeyword",
+        "pageTitleLength",
+        "firstParagraph",
+        "keywordDoubles"];
+    this.replacedVars = {};
+    this.getData();
+};
+
+/**
+ * Get data from inputfields and store them in an analyzerData object. This object will be used to fill
+ * the analyzer and the snippetpreview
+ */
+YoastSEO_WordPressScraper.prototype.getData = function() {
+    this.analyzerData.keyword = this.getDataFromInput( "keyword" );
+    this.analyzerData.meta = this.getDataFromInput( "meta" );
+    this.analyzerData.text = this.getDataFromInput( "text" );
+    this.analyzerData.pageTitle = this.getDataFromInput( "pageTitle" );
+    this.analyzerData.title = this.getDataFromInput( "title" );
+    this.analyzerData.url = this.getDataFromInput( "url" );
+    this.analyzerData.excerpt = this.getDataFromInput( "excerpt" );
+    this.analyzerData.snippetTitle = this.getDataFromInput( "snippetTitle" );
+    this.analyzerData.snippetMeta = this.getDataFromInput( "meta" );
+    this.analyzerData.snippetCite = this.getDataFromInput( "cite" );
+};
+
+/**
+ * gets the values from the given input. Returns this value
+ * @param inputType
+ * @returns value
+ */
+YoastSEO_WordPressScraper.prototype.getDataFromInput = function( inputType ) {
+    var val;
+    switch( inputType){
+        case "text":
+        case "content":
+            val = this.getContentTinyMCE();
+            break;
+        case "url":
+			val = document.getElementById("sample-permalink").innerHTML.split("<span")[0];
+			break;
+        case "cite":
+		case "editable-post-name":
+			val = document.getElementById("editable-post-name").textContent;
+			var elem = document.getElementById("new-post-slug");
+			if(elem !== null && val === "") {
+				val = document.getElementById("new-post-slug").value;
+			}
+            break;
+        case "meta":
+            val = document.getElementById("yoast_wpseo_metadesc").value;
+            break;
+        case "keyword":
+            val = document.getElementById("yoast_wpseo_focuskw").value;
+            break;
+        case "title":
+        case "snippetTitle":
+            val = document.getElementById("yoast_wpseo_title").value;
+            break;
+        case "pageTitle":
+             val = document.getElementById("title").value;
+            break;
+        case "excerpt":
+            val = document.getElementById("excerpt").value;
+            break;
+        default:
+            break;
+    }
+    return val;
+};
+
+/**
+ * When the snippet is updated, update the (hidden) fields on the page
+ * @param value
+ * @param type
+ */
+YoastSEO_WordPressScraper.prototype.setDataFromSnippet = function( value, type) {
```

Comment:
Here applies the same as above, we should just have `setSnippetMeta`, `setSnippetCite`, `setSnippetTitle`.